site/gatsby-site/server/context.ts

```diff
@@ -56,35 +51,75 @@ export const verifyToken = async (token: string) => {
 }
 
 async function getUser(userId: string, client: MongoClient) {
-
     const db = client.db('customData');
-
     const collection = db.collection('users');
-
     const userData = await collection.findOne({ userId });
 
     return {
         id: userId,
         roles: userData?.roles,
-    }
+    };
 }
 
-async function getUserFromHeader(header: string, client: MongoClient) {
+async function getTokenCache(token: string, client: MongoClient) {
+    const db = client.db('customData');
+    const collection = db.collection('tokenCache');
```

Comment:
drive-by: let's discuss how to cache tokens before creating a DB collection for this.